### DISCUSSION TITLE
Query two MongoDB collections using Hot Chocolate MongoDB NuGet.

### DISCUSSION THREAD

**Discussion by mddheeraj** [ChilliCream/hotchocolate#3449] [Q&A]
Hi,

We are implementing a GraphQL server that needs to get data from MongoDB from the backend. We found that hot chocolate has released a NuGet which supports MongoDB which is really helpful.  
The documentation shows querying a single collection but we need to query more than one collection.
Do you have sample documentation to query more than one collection and return results from two collections, like joining two tables?

Thank You,
Dheeraj

**Comment by PascalSenn** [comment]
Hi @mddheeraj  

I'm not sure if I understand this correctly :) Can you write a graphql query / json response to show what you exactly mean? :) 

Like
```graphql
query Foo 
{
     users {  # collection a
          addresses  {  #collection n
                 street
          }
      }
}

**Reply by mddheeraj** [reply]
How the Query.cs looks like? What all changes to be done in the Query.cs to query two collections? 
At present in the [Query.cs](https://github.com/ChilliCream/hotchocolate-examples/blob/master/misc/MongoDB/Query.cs), the executable returned is specific to one collection(Person) but want to query two collections.

**Reply by PascalSenn** [reply]
@mddheeraj  what do you want to return from the resolver then? can you share some code? 
even pseudo code would be good

**Reply by mddheeraj** [reply]
Same as the one you mentioned. According to the query users and addresses are separate collections in MongoDB. What all changes to be done in code?

``` 
query Foo 
{
     users {  # collection a
          addresses  {  #collection n
                 street
          }
      }
}
```

**Reply by mddheeraj** [reply]
Hi @PascalSenn any luck or sample code for the above query.

**Reply by PascalSenn** [reply]
hi @mddheeraj 

You  would use just use two resolvers for this

```csharp
[ExtendObjectType("Query")]
public class Query 
{
     public IList<User> GetUsers([Service] YourRepo repo) => repo.GetUsers();
}

[ExtendObjectType(nameof(User))]
public class UserExtensions 
{
     public IList<Address> GetAddresses(
         [Parent] User user,
         [Service] YourRepo repo) => repo.GetAddressesOfUser(user.Id);
}


services.AddGraphQLServer()
    .AddQueryType(x => x.Name("Query"))
    .AddType<Query>()
    .Addtype<UserExtensions>();

```

To work around the N+1 issue of the GetAddresses resolver, you probably want to use a dataloader
You can find a complete example of there here: 
https://github.com/ChilliCream/graphql-workshop

**Reply by mddheeraj** [reply]
Thanks @PascalSenn for quick reply. I’ll dig into the workshop.
